docs/devel/pull-requests.md

```diff
@@ -73,16 +73,12 @@ The following will save time for both you and your reviewer:
 
 ## Release Notes
 
-All pull requests are initiated with a `needs-release-note` label.
-There are many `release-note-*` label options, including `release-note-none`.
-If your PR does not require any visibility at release time, you may use a
-`release-note-none` label.  Otherwise, please choose a `release-note-*` label
-that fits your PR.
-
-Additionally, `release-note-none` is not allowed on PRs on release branches.
-
-Finally, ensure your PR title is the release note you want published at relase
-time.
+1. Your PR title is the **release note** you want published at release time.
```

Comment:
There are 2 different kinds of releases.  There are major new branch releases and then there's everything else (alpha, beta, patch).  Automated release notes are not meant to target the major new branch releases (as the example you gave).  No amount of automation is going to construct lengthy, detailed and easy to read descriptions that were hand crafted as in the example of 1.2.0.

As I pointed out in the proposal, the automation should aim to provide automated release notes for every kind of release except the major new branch kind and for those the automation is merely bootstrapping the release notes so that the release owner can make them look like the carefully constructed release detail in your 1.2.0 example.

Another thing to consider is the workflow and how frictionless release note collection is to contributors.  The PR title is something people are used to providing already and is the current workflow.  Asking them to further be sure to construct a parsable, lengthy release note somewhere in the body of each PR is really putting a burden on folks, IMO.  And chances are they won't get it right and there will be a significant amount of editing anyway.
Furthermore, there's no built-in way (using git or github) to validate the body by looking for a tag and a description, though I suppose a custom bot could be designed.

The current solution gets us to ~90% with automated notes for alpha, beta, minor and bootstrapping of major releases.  If you could say the additional work of asking contributors to fill in a "good release note" in the body somewhere, parsing it, extracting and formatting it gets us another 5%, I'm not sure that's worth it and like I said before it could end up creating more work in the end for the branch owner to pull apart the mess of title + additional body contents/formatting.

If we do want to investigate this, it's certainly worth it's own issue.  There will be lots of work to do to make that happen and it will take time to implement and test.